ga4gh/datamodel/__init__.py

```diff
@@ -56,6 +81,12 @@ class PysamDatamodelMixin(object):
 
     maxStringLength = 2**10  # arbitrary
 
+    # LRU cache of open file handles
+    cache = PysamFileHandleCache()
```

Comment:
The other class-level attributes of the mixin are immutable; I'm not sure if adding mutable class-level attributes is going to give us one file cache or N file caches (where N = the number of instantiations of objects using the mixin).  Pretty sure it's the former, but we should confirm that in a test to be sure we are getting the desired behavior.

I guess what bugs me about this pattern is that we are combining the singleton pattern (cache) with the mixin pattern.  Separation of concerns seems to say that they should be distinct... @jeromekelleher what do you think?